core/src/main/java/tachyon/worker/BlockHandlerLocal.java

```diff
@@ -84,25 +85,33 @@ public boolean delete() throws IOException {
   }
 
   @Override
-  public ByteBuffer read(long blockOffset, int length) throws IOException {
+  public ByteChannel getChannel() {
+    return mLocalFileChannel;
+  }
+
+  @Override
+  public long getLength() throws IOException {
+    return mLocalFile.length();
+  }
+
+  @Override
+  public ByteBuffer read(long offset, int length) throws IOException {
     long fileLength = mLocalFile.length();
     String error = null;
-    if (blockOffset > fileLength) {
-      error =
-          String.format("blockOffset(%d) is larger than file length(%d)", blockOffset, fileLength);
-    }
-    if (error == null && length != -1 && blockOffset + length > fileLength) {
+    if (offset > fileLength) {
+      error = String.format("offset(%d) is larger than file length(%d)", offset, fileLength);
+    } else if (length != -1 && offset + length > fileLength) {
       error =
-          String.format("blockOffset(%d) plus length(%d) is larger than file length(%d)",
-              blockOffset, length, fileLength);
+          String.format("offset(%d) plus length(%d) is larger than file length(%d)", offset,
+              length, fileLength);
     }
     if (error != null) {
       throw new IllegalArgumentException(error);
```

Comment:
Would this code be called by worker? If that's the case, would this crash the worker? Is it reasonable to crash the worker by this?